### ISSUE TITLE
How to improve Accuracy of YOLO11n-cls on Medical Image Classification (Tumor Type Classification)

### ISSUE THREAD

**Issue by Helya-Haji** [ultralytics/ultralytics#22715]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello,

I’m working on a **medical image classification** project (**tumor type classification**) using **YOLO11n-cls**, and I’m getting good performance but the accuracy seems to have reached a plateau.

My current metrics:
Validation accuracy: ~96%
Test accuracy: ~94%

I would like to improve the model further and was wondering what techniques you recommend specifically for medical image classification, where subtle features and fine-grained textures are important.

Are there recommended techniques or best practices for improving accuracy of YOLO11n-cls on medical datasets?

Is there a suggested way to adjust the backbone or feature extraction for more fine-grained features?

Is there any official guidance or examples for training YOLO-cls models on medical image datasets?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Helya-Haji, thank you for your detailed question and for using Ultralytics YOLO11n-cls on such an important medical task 🙌 This is an **automated** response to help you get started quickly — an Ultralytics engineer will also review your issue and follow up soon 👨‍💻👩‍💻

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many common questions may already be answered.

If this is a 🐛 Bug Report (unexpected behavior, crashes, or inconsistent results), please also provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate more effectively.

Since your post is a custom training ❓ Question, it would be very helpful if you could share a bit more detail so the team can give more targeted guidance, for example:
- Your training command or Python script (with key args like `model`, `data`, `imgsz`, `epochs`, `lr0`, `batch`, etc.)
- A brief description of your dataset (number of classes, class distribution, image resolution, color/grayscale, any preprocessing)
- A few example images (or crops) showing typical tumor types and edge cases
- Training logs and learning curves (loss, accuracy, confusion matrix, any signs of over/underfitting)
- Any custom augmentations, class weights, or modifications you’ve already tried

In the meantime, you can also review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure your current setup follows recommended training practices ✅

Join the Ultralytics community where it suits you best:
- For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧
- For longer-form technical discussions, visit [Discourse](https://community.ultralytics.com/)
- To explore and share community experiences, check out our [Subreddit](https://www.reddit.com/r/Ultralytics/)  

## Upgrade

Before diving deeper, please ensure you are on the latest `ultralytics` version (this can sometimes improve training behavior and metrics). Upgrade in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) and the latest [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml):

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You should use a larger model if you want better accuracy, like `yolo11m-cls.pt` or `yolo11l-cls.pt`

**Comment by Helya-Haji** [comment]
I actually tried using yolo11l-cls as well, but the model doesn’t learn at all.
Its accuracy gets worse with every epoch, and I’ve tested it several times with different settings, but the results are still the same.

@Y-T-G

**Comment by Y-T-G** [comment]
How many images do you have and what's your training command?

**Comment by Helya-Haji** [comment]
This is my training script:

```python
from ultralytics import YOLO
from pathlib import Path
import torch

# Initialize model
model = YOLO(r"E:\university\project\yolo11n-cls.pt")

# Train the model
results = model.train(
    data=r"E:\university\project\New folder\UWF dataset",  
    epochs=100,
    imgsz=224,
    batch=16,
    lr0=0.001,
    augment=True,
    cos_lr =True,
    device=1 if torch.cuda.is_available() else 'cpu',
    project=r"E:\university\project\models\YOLO",
    name="yolo11_original_dataset",
    save_period=10,
    amp=True
)

print("Training completed!")
print(f"Best model saved at: {results.save_dir}/weights/best.pt")
```


and I have 2000 total images for 6 classes.

@Y-T-G

**Comment by Y-T-G** [comment]
Remove

```
augment=True,
cos_lr =True,
```

**Comment by Helya-Haji** [comment]
I tried that. The accuracy of training increased 1% but the kappa-score and the F1-score decreased.
Moreover, the accuracy on test set decreased to 92% 

@Y-T-G

**Comment by Y-T-G** [comment]
Which model did you train?  You should train `yolo11l-cls.pt` without those arguments

**Comment by Helya-Haji** [comment]
I trained yolo11l-cls without them. It increased test accuracy 1% but it did not increased validation accuracy.
@Y-T-G
